modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -30,6 +30,27 @@ RadialReturnStressUpdateTempl<is_ad>::validParams()
       "effective_inelastic_strain_name",
       "Name of the material property that stores the effective inelastic strain");
   params.addParam<bool>("use_substep", false, "Whether to use substepping");
+
+  MooseEnum substeppingType("NONE ERROR_BASED INCREMENT_BASED", "NONE");
+  substeppingType.addDocumentation("NONE", "Do not use substepping");
+  substeppingType.addDocumentation(
+      "ERROR_BASED",
+      "Use substepping with a substep size that will yield, at most, the creep numerical "
+      "integration error given by substep_strain_tolerance.");
+  substeppingType.addDocumentation(
+      "INCREMENT_BASED",
+      "Use substepping with a substep size that will keep each inelastic strain increment below "
+      "the maximum inelastic strain increment allowed in a time step.");
+  params.addParam<MooseEnum>(
+      "use_substepping", substeppingType, "Whether and how to use substepping");
+  params.addParam<bool>(
+      "adaptive_substepping",
+      false,
+      "Use adaptive substepping, where the number of substeps is successively doubled until the "
+      "return mapping model successfully converges or the maximum number of substeps is reached. ");
+
+  params.addDeprecatedParam<bool>(
+      "use_substep", false, "Whether to use substepping", "Use `use_substepping` instead");
```

Comment:
Are we doing the deadline thing to remove deprecated parameters? I cannot see a reason to keep this one for long.